src/main/java/org/alliancegenome/curation_api/services/AlleleService.java

```diff
@@ -76,60 +72,40 @@
 	@Override
 	@Transactional
 	public ObjectResponse<Allele> deleteById(Long id) {
-		removeOrDeprecateNonUpdated(id, "Allele DELETE API call");
+		deprecateOrDelete(id, true, "Allele DELETE API call", false);
 		ObjectResponse<Allele> ret = new ObjectResponse<>();
 		return ret;
 	}
 
+	@Override
 	@Transactional
-	public void removeOrDeprecateNonUpdated(Long id, String loadDescription) {
+	public Allele deprecateOrDelete(Long id, Boolean throwApiError, String requestSource, Boolean forceDeprecate) {
 		Allele allele = alleleDAO.find(id);
 		if (allele != null) {
-			List<Long> referencingDAIds = alleleDAO.findReferencingDiseaseAnnotationIds(id);
-			Boolean anyReferencingEntities = false;
-			for (Long daId : referencingDAIds) {
-				DiseaseAnnotation referencingDA = diseaseAnnotationService.deprecateOrDeleteAnnotationAndNotes(daId, false, loadDescription, true);
-				if (referencingDA != null) {
-					anyReferencingEntities = true;
-				}
-			}
-
-			List<Long> referencingPAIds = alleleDAO.findReferencingPhenotypeAnnotations(id);
-			for (Long paId : referencingPAIds) {
-				PhenotypeAnnotation referencingPA = phenotypeAnnotationService.deprecateOrDeleteAnnotationAndNotes(paId, false, loadDescription, true);
-				if (referencingPA != null) {
-					anyReferencingEntities = true;
-				}
-			}
-			if (CollectionUtils.isNotEmpty(allele.getAlleleGeneAssociations())) {
-				for (AlleleGeneAssociation association : allele.getAlleleGeneAssociations()) {
-					association = alleleGeneAssociationService.deprecateOrDeleteAssociation(association.getId(), false, loadDescription, true);
-					if (association != null) {
-						anyReferencingEntities = true;
-					}
-				}
-			}
-			if (CollectionUtils.isNotEmpty(allele.getConstructGenomicEntityAssociations())) {
-				for (ConstructGenomicEntityAssociation association : allele.getConstructGenomicEntityAssociations()) {
-					association = constructGenomicEntityAssociationService.deprecateOrDeleteAssociation(association.getId(), false, loadDescription, true);
-					if (association != null) {
-						anyReferencingEntities = true;
-					}
-				}
-			}
-			if (anyReferencingEntities) {
+			if (forceDeprecate || alleleDAO.hasReferencingDiseaseAnnotationIds(id) || alleleDAO.hasReferencingPhenotypeAnnotations(id) ||
+					CollectionUtils.isNotEmpty(allele.getAlleleGeneAssociations()) ||
```

Comment:
🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.whitespace.OperatorWrapCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'||' should be on a new line.